ext/xmlwriter/php_xmlwriter.stub.php

```diff
@@ -6,85 +6,85 @@ function xmlwriter_open_uri(string $uri): XMLWriter|false {}
+function xmlwriter_write_element(XMLWriter $writer, string $name, ?string $content = null): bool {}
 
-function xmlwriter_write_element_ns(XMLWriter $xmlwriter, ?string $prefix, string $name, ?string $uri, ?string $content = null): bool {}
+function xmlwriter_write_element_ns(XMLWriter $writer, ?string $prefix, string $name, ?string $uri, ?string $content = null): bool {}
 
-function xmlwriter_start_pi(XMLWriter $xmlwriter, string $target): bool {}
+function xmlwriter_start_pi(XMLWriter $writer, string $target): bool {}
 
-function xmlwriter_end_pi(XMLWriter $xmlwriter): bool {}
+function xmlwriter_end_pi(XMLWriter $writer): bool {}
 
-function xmlwriter_write_pi(XMLWriter $xmlwriter, string $target, string $content): bool {}
+function xmlwriter_write_pi(XMLWriter $writer, string $target, string $content): bool {}
 
-function xmlwriter_start_cdata(XMLWriter $xmlwriter): bool {}
+function xmlwriter_start_cdata(XMLWriter $writer): bool {}
 
-function xmlwriter_end_cdata(XMLWriter $xmlwriter): bool {}
+function xmlwriter_end_cdata(XMLWriter $writer): bool {}
 
-function xmlwriter_write_cdata(XMLWriter $xmlwriter, string $content): bool {}
+function xmlwriter_write_cdata(XMLWriter $writer, string $content): bool {}
 
-function xmlwriter_text(XMLWriter $xmlwriter, string $content): bool {}
+function xmlwriter_text(XMLWriter $writer, string $content): bool {}
 
-function xmlwriter_write_raw(XMLWriter $xmlwriter, string $content): bool {}
+function xmlwriter_write_raw(XMLWriter $writer, string $content): bool {}
 
-function xmlwriter_start_document(XMLWriter $xmlwriter, ?string $version = "1.0", ?string $encoding = null, ?string $standalone = null): bool {}
+function xmlwriter_start_document(XMLWriter $writer, ?string $version = "1.0", ?string $encoding = null, ?string $standalone = null): bool {}
 
-function xmlwriter_end_document(XMLWriter $xmlwriter): bool {}
+function xmlwriter_end_document(XMLWriter $writer): bool {}
 
-function xmlwriter_write_comment(XMLWriter $xmlwriter, string $content): bool {}
+function xmlwriter_write_comment(XMLWriter $writer, string $content): bool {}
 
-function xmlwriter_start_dtd(XMLWriter $xmlwriter, string $qualifiedName, ?string $publicId = null, ?string $systemId = null): bool {}
+function xmlwriter_start_dtd(XMLWriter $writer, string $qualified_name, ?string $public_id = null, ?string $system_id = null): bool {}
 
-function xmlwriter_end_dtd(XMLWriter $xmlwriter): bool {}
+function xmlwriter_end_dtd(XMLWriter $writer): bool {}
 
-function xmlwriter_write_dtd(XMLWriter $xmlwriter, string $name, ?string $publicId = null, ?string $systemId = null, ?string $subset = null): bool {}
+function xmlwriter_write_dtd(XMLWriter $writer, string $name, ?string $public_id = null, ?string $system_id = null, ?string $content = null): bool {}
 
-function xmlwriter_start_dtd_element(XMLWriter $xmlwriter, string $qualifiedName): bool {}
+function xmlwriter_start_dtd_element(XMLWriter $writer, string $qualified_name): bool {}
```

Comment:
As these are aliased with `XmlWriter::startDtdElement()` etc and have common documentation (https://www.php.net/manual/en/function.xmlwriter-start-dtd-element.php), I believe we need the parameter names to match as well. Is that correct @cmb69?

Reply:
Oops, I was the one who told Dik that it's possible to use different names for the procedural and the OO API (since it has no technical limitation). Unfortunately, I didn't think about the documentation :/

Reply:
As it is now, the procedural and OOP API have individual `<methodsynopsis>` elements, so this is not an issue for the documentation. I am generally not sure that having different parameter names for dual APIs is a good idea, though, since that may make transitioning from procedural to OOP slightly harder (the other way round is equally harder, but likely not a relevant issue).

Reply:
@cmb69 The issue I had in mind here is that parameter names are also referenced in the body of the documentation, not just the method synopsis. There wouldn't be a unique parameter name to use in that case (though I guess using either one wouldn't be terrible either).

Reply:
Oh, indeed, that wouldn't be nice.

Reply:
@kocsismate I did not realize that either, even though I had my eyes on that documentation while preparing this PR. :)

It would be a shame if this means we cannot adhere to naming conventions. In the past the mysqli extension (and probably others as well) failed to get this right for the method / function names. Now we have to live with that. We are at a similar decision point for parameter names. Do it right or live with it.

The documentation currently shows both variants of the function names and signatures. The parameter list shows just one. Unfortunately I have no idea how hard it would be to fix that and if this must be fixed before the release date or not...

Reply:
Do the docs currently have different param names for procedural and OOP?

Reply:
@cmb69 Well, at least they currently show separate signatures. Although I have not seen any examples yet I _assume_ that these signatures could show different parameter names?

Reply:
As said above, yes, there can be different signatures, but the param descriptions can not (easily) be duplicated.

Reply:
> @cmb69 The issue I had in mind here is that parameter names are also referenced in the body of the documentation, not just the method synopsis. There wouldn't be a unique parameter name to use in that case (though I guess using either one wouldn't be terrible either).

I do agree that using either one would be an acceptable price to pay. Also, the PRs do tend to make the parameter names shorter. That means fewer parameters that actually differ between the two APIs.

Reply:
As it is now, the parameter names are clickable (anchor links for JS only); this wouldn't work with additional info in the `<parameter>` elements. However, `<varlistentry>` elements allow multiple `<term>`s, so we could write it that way. "Only" PhD would need to be adapted (it apparently renders the first `<term>` only).

Reply:
After some further consideration, I somehow like the idea of multiple `<term><parameter></parameter></term>` elements, but I'm somewhat scared to write or read `<parameter>myArg</parameter>/<parameter>my_arg</parameter>`.

Reply:
FYI: The same problem affects ext/date too (I'm not aware of any more prominent dual API provider extension), but currently it defines all the procedural params as camelCase. I realized this a few days ago.

Reply:
So, while I'm sure there's a way to solve the technical docbook/phd problem here, I'm not convinced that it really makes sense to have different parameter names in the procedural and OO APIs. If nothing else, it makes the documentation more complex and harder to understand, for an issue that really nobody cares about.

My view here is that for these kinds of mixed procedural/OO APIs the procedural API is just a legacy leftover that should ideally be removed at some point, and we should just use the same names as in the OO interface there. (However, there *is* a potential complication: If want to add an OO mirror API for an existing procedural API at a later point in time, we'd end up using the original procedural names.)

I think it is worth considering to simply **always** use `$snake_case` style parameter names inside php-src. This will make the style consistent for all bundled extensions, no matter what kind of API they expose. It looks like PSR-12 does not prescribe a particular casing style for variables (but maybe I missed it). Though I do admit that $camelCase seems to be preferred in userland.

Reply:
(Of course, always using $camelCase in php-src is also an option, I just suggested snake_case as it seems to be currently predominant.)

Reply:
> I think it is worth considering to simply **always** use `$snake_case` style parameter names inside php-src.

Does this not go against the general direction to make internals and userland more alike, blend more naturally? At some point, internal APIs may even be written in PHP. In that scenario, having internal APIs look less alien from a userland perspective seems relevant...

> (Of course, always using $camelCase in php-src is also an option, I just suggested snake_case as it seems to be currently predominant.)

I prefer that proposal. I agree that the procedural APIs are of lesser importance. Making the OO APIs come out right is the main thing. If the procedural versions also come out right, that is a nice to have.

Reply:
> (Of course, always using $camelCase in php-src is also an option, I just suggested snake_case as it seems to be currently predominant.)

Just to make sure I understand your proposal correctly:

- Pure OO APIs: camelCase
- Dual APIs: camelCase
- Pure procedural APIs: snake_case

Reply:
Yes, either that, or pick one of camelCase/snake_case for everything.

Reply:
I think keeping the snake_case for pure procedural is nice. It makes the parameter naming style match the existing snake_case function names. Also, most APIs are in that style already.

Using camelCase for everything else nicely matches the style of most userland code.

Reply:
Just pushed an update to apply Nikita's proposal.